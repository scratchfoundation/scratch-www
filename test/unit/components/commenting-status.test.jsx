const React = require('react');
const {shallow} = require('enzyme');
const CommentingStatus = require('../../../src/components/commenting-status/commenting-status.jsx');

describe('CommentingStatus', () => {
    test('Basic render', () => {
        const component = shallow(
            <CommentingStatus />
        );
        expect(component.find('div.commenting-status').exists()).toBe(true);
        expect(component.find('img.comment-status-icon').exists()).toBe(true);
    });

    test('ClassNames added', () => {
        const component = shallow(
            <CommentingStatus
                className="class1"
                innerClassName="class2"
            />
        );
        expect(component.find('div.class1').exists()).toBe(true);
        expect(component.find('div.class2').exists()).toBe(true);
    });

    test('Children added', () => {
        const component = shallow(
            <CommentingStatus>
                <img className="myChildDiv" />
            </CommentingStatus>
        );
        expect(component.find('img.myChildDiv').exists()).toBe(true);
    });
});
