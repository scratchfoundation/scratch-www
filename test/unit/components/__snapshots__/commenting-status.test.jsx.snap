// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentingStatus Basic render 1`] = `
<div
  class="commenting-status"
>
  <div
    class="commenting-status-inner-content"
  >
    <div
      class="flex-row comment-status-img"
    >
      <img
        class="comment-status-icon"
        src="/svgs/project/comment-status.svg"
      />
    </div>
    <div
      class="flex-row"
    />
  </div>
</div>
`;

exports[`CommentingStatus Children added 1`] = `
<div
  class="commenting-status"
>
  <div
    class="commenting-status-inner-content"
  >
    <div
      class="flex-row comment-status-img"
    >
      <img
        class="comment-status-icon"
        src="/svgs/project/comment-status.svg"
      />
    </div>
    <div
      class="flex-row"
    >
      <img
        class="myChildDiv"
      />
    </div>
  </div>
</div>
`;

exports[`CommentingStatus ClassNames added 1`] = `
<div
  class="commenting-status class1"
>
  <div
    class="commenting-status-inner-content class2"
  >
    <div
      class="flex-row comment-status-img"
    >
      <img
        class="comment-status-icon"
        src="/svgs/project/comment-status.svg"
      />
    </div>
    <div
      class="flex-row"
    />
  </div>
</div>
`;
